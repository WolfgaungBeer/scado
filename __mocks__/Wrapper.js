import React from 'react';
import { Provider } from 'react-redux';
import configureStore from 'redux-mock-store';
import { Theme, defaultTheme } from '../scado';

const mockStore = configureStore();
const store = mockStore({
    scado: {
        theme: defaultTheme,
        toasts: [
            { id: '1', type: 'success', content: 'Text', onClose: () => {} },
            { id: '2', type: 'error', content: 'Text', onClose: () => {} },
        ],
    },
});


const Wrapper = ({ children }) => ( // eslint-disable-line
    <Provider store={store}>
        <Theme>{children}</Theme>
    </Provider>
);

export default Wrapper;
