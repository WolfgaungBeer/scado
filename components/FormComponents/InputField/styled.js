import styled from 'styled-components';

export const FieldWrapper = styled.div`
    padding: 0.25rem 0 1rem 0;
`;

export const FieldLabel = styled.label`
    width: 100%;
    color: ${props => props.theme.color.black};
`;

export const FieldInput = styled.input`
    margin: 0.25rem 0 0 0;
    box-sizing: border-box;
    width: 100%;
    font-size: 1.2rem;
    line-height: 1.8rem;
    border-style: solid;
    border-width: 1px;
    border-color: ${props => props.theme.color.lightGray};
    border-radius: 4px;
`;
