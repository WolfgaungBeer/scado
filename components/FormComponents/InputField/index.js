import React from 'react';
import { oneOf, node, bool } from 'prop-types';
import fieldPropTypes from '../fieldPropTypes';
import defaultFieldPropTypes from '../defaultFieldPropTypes';
import { FieldWrapper, FieldLabel, FieldInput } from './styled';
import Icon from '../../Icon';
import Text from '../../Text';

const propTypes = {
    ...fieldPropTypes,
    type: oneOf(['email', 'text', 'password', 'number', 'url']).isRequired,
    label: node,
    required: bool,
};

const defaultProps = {
    ...defaultFieldPropTypes,
    type: undefined,
    label: undefined,
    required: undefined,
};

const InputField = ({ input, meta, type, label, required }) => {
    const { touched, error, warning, form } = meta;
    const hasError = (touched && error);
    const hasWarning = (touched && warning);
    const id = `${form}-${input.name}`;
    return (
        <FieldWrapper>
            {label &&
                <FieldLabel htmlFor={id}>{label} {required && <Text color="error">*</Text>}</FieldLabel>}
            <FieldInput id={id} type={type} {...input} />
            {hasError &&
                <Text color="error" scale="s"><Icon icon="clear" color="error" scale="s" /> {error}</Text>}
            {hasWarning &&
                <Text color="warning" scale="s"><Icon icon="warning" color="warning" scale="s" /> {warning}</Text>}
        </FieldWrapper>
    );
};

InputField.propTypes = propTypes;
InputField.defaultProps = defaultProps;

export default InputField;
