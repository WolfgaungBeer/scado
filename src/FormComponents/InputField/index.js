import React from 'react';
import { oneOf, node, bool } from 'prop-types';
import fieldPropTypes from '../fieldPropTypes';
import defaultFieldPropTypes from '../defaultFieldPropTypes';
import { FieldWrapper, FieldInput } from './styled';
import Icon from '../../Icon';
import Text from '../../Text';

const propTypes = {
    ...fieldPropTypes,
    type: oneOf(['email', 'text', 'password', 'number', 'url']).isRequired,
    label: node,
    required: bool,
};

const defaultProps = {
    ...defaultFieldPropTypes,
    type: undefined,
    label: undefined,
    required: undefined,
};

const InputField = ({ input, meta, type, label, required }) => {
    const { touched, error, warning, form } = meta;
    const hasError = (touched && error);
    const hasWarning = (touched && warning);
    const id = `${form}-${input.name}`;
    return (
        <FieldWrapper>
            {label &&
                <Text.Label htmlFor={id}>{label} {required && <Text.Span color="error">*</Text.Span>}</Text.Label>}
            <FieldInput id={id} type={type} {...input} />
            {hasError &&
                <Text.Span color="error" scale="s"><Icon color="error" scale="s">clear</Icon> {error}</Text.Span>}
            {hasWarning &&
                <Text.Span color="warning" scale="s"><Icon color="warning" scale="s">warning</Icon> {warning}</Text.Span>}
        </FieldWrapper>
    );
};

InputField.propTypes = propTypes;
InputField.defaultProps = defaultProps;

export default InputField;
