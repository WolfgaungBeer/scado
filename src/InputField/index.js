import React from 'react';
import { oneOf, node, bool, string, func } from 'prop-types';
import fieldPropTypes from '../fieldPropTypes';
import defaultFieldPropTypes from '../defaultFieldPropTypes';
import { FieldWrapper, FieldInput } from './styled';
import Icon from '../Icon';
import Text from '../Text';

const propTypes = {
    ...fieldPropTypes,
    type: oneOf(['email', 'text', 'password', 'number', 'url']).isRequired,
    label: node,
    required: bool,
    className: string,
    onClick: func,
};

const defaultProps = {
    ...defaultFieldPropTypes,
    type: undefined,
    label: undefined,
    required: undefined,
    className: undefined,
    onClick: undefined,
};

const InputField = ({ input, meta, type, label, required, className, onClick }) => {
    const { touched, error, warning, form } = meta;
    const hasError = (touched && error);
    const hasWarning = (touched && warning);
    const id = `${form}-${input.name}`;
    return (
        <FieldWrapper className={className} onClick={onClick}>
            {label &&
                <Text.Label htmlFor={id}>{label} {required && <Text.Span color="error">*</Text.Span>}</Text.Label>}
            <FieldInput id={id} type={type} {...input} />
            {hasError &&
                <Text.Span color="error" scale="s"><Icon color="error" scale="s">clear</Icon> {error}</Text.Span>}
            {hasWarning &&
                <Text.Span color="warning" scale="s"><Icon color="warning" scale="s">warning</Icon> {warning}</Text.Span>}
        </FieldWrapper>
    );
};

InputField.propTypes = propTypes;
InputField.defaultProps = defaultProps;

export default InputField;
