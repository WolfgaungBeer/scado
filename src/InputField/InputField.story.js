import React from 'react';
import { storiesOf } from '@storybook/react';
import { text, boolean } from '@storybook/addon-knobs';
import InputField from './';

const errorMeta = { touched: true, error: 'This Field has an error' };
const warningMeta = { touched: true, warning: 'This Field has an warning' };

storiesOf('InputField', module)
    .add('default', () => (
        <InputField type="text" label="InputField Example" />
    ))
    .add('with error', () => (
        <InputField type="text" label="InputField Example" meta={errorMeta} />
    ))
    .add('with warning', () => (
        <InputField type="text" label="InputField Example" meta={warningMeta} />
    ))
    .add('with props', () => (
        <InputField
            type={text('type', 'text')}
            label={text('label', 'InputField')}
            required={boolean('required', false)}
        />
    ));
