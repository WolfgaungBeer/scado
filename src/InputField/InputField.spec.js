import React from 'react';
import renderer from 'react-test-renderer';
import Wrapper from '../../__mocks__/Wrapper';
import InputField from './';

const errorMeta = { touched: true, error: 'This Field has an error' };
const warningMeta = { touched: true, warning: 'This Field has an warning' };

it('renders the Icon Component correctly', () => {
    const tree1 = renderer.create(
        <Wrapper>
            <InputField type="text" />
        </Wrapper>,
    ).toJSON();
    expect(tree1).toMatchSnapshot();

    const tree2 = renderer.create(
        <Wrapper>
            <InputField type="text" meta={errorMeta} />
        </Wrapper>,
    ).toJSON();
    expect(tree2).toMatchSnapshot();

    const tree3 = renderer.create(
        <Wrapper>
            <InputField type="text" meta={warningMeta} />
        </Wrapper>,
    ).toJSON();
    expect(tree3).toMatchSnapshot();

    const tree4 = renderer.create(
        <Wrapper>
            <InputField type="text" required label="A InputField" />
        </Wrapper>,
    ).toJSON();
    expect(tree4).toMatchSnapshot();
});
