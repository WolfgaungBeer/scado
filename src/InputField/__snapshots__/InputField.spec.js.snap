// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders the Icon Component correctly 1`] = `
<div
  className="sc-EHOje cOnjOH"
>
  <input
    className="sc-bZQynM fZBtPE"
    id="undefined-"
    name=""
    onBlur={undefined}
    onChange={undefined}
    onDragStart={undefined}
    onDrop={undefined}
    onFocus={undefined}
    type="text"
    value=""
  />
</div>
`;

exports[`renders the Icon Component correctly 2`] = `
<div
  className="sc-EHOje cOnjOH"
>
  <input
    className="sc-bZQynM fZBtPE"
    id="undefined-"
    name=""
    onBlur={undefined}
    onChange={undefined}
    onDragStart={undefined}
    onDrop={undefined}
    onFocus={undefined}
    type="text"
    value=""
  />
  <span
    className="sc-jzJRlG cMTDVO"
    color="error"
    scale="s"
  >
    <i
      className="material-icons  sc-ifAKCX eUJGIr"
      color="error"
      onClick={undefined}
      scale="s"
    >
      clear
    </i>
     
    This Field has an error
  </span>
</div>
`;

exports[`renders the Icon Component correctly 3`] = `
<div
  className="sc-EHOje cOnjOH"
>
  <input
    className="sc-bZQynM fZBtPE"
    id="undefined-"
    name=""
    onBlur={undefined}
    onChange={undefined}
    onDragStart={undefined}
    onDrop={undefined}
    onFocus={undefined}
    type="text"
    value=""
  />
  <span
    className="sc-jzJRlG iXudPJ"
    color="warning"
    scale="s"
  >
    <i
      className="material-icons  sc-ifAKCX cWoMMb"
      color="warning"
      onClick={undefined}
      scale="s"
    >
      warning
    </i>
     
    This Field has an warning
  </span>
</div>
`;

exports[`renders the Icon Component correctly 4`] = `
<div
  className="sc-EHOje cOnjOH"
>
  <label
    className="sc-fjdhpX eKDuiA"
    htmlFor="undefined-"
  >
    A InputField
     
    <span
      className="sc-jzJRlG cAQGLB"
      color="error"
    >
      *
    </span>
  </label>
  <input
    className="sc-bZQynM fZBtPE"
    id="undefined-"
    name=""
    onBlur={undefined}
    onChange={undefined}
    onDragStart={undefined}
    onDrop={undefined}
    onFocus={undefined}
    type="text"
    value=""
  />
</div>
`;
